### ISSUE TITLE
Forge crashing when essentials is in mod directory

### ISSUE THREAD

**Issue by Shanehar** [Da-Technomancer/Essentials#15]
Tried both essentials-1.14.2-2.1.1.jar and essentials-1.14.2-2.1.0.jar. Using forge version forge-1.14.2-26.0.63.jar. freshly installed. Mincrafter server version minecraft_server.1.14.2.jar on debian 9. 

I've tested other random mods with no issue. If i run forge modless or with other mods it launches fine. If the essentials Jar file is in the mods directory, forge won't load. here is the output on the screen when it fails. 

```
oot@tbot:/home/minecraft#  java -jar forge-1.14.2-26.0.63.jar
[22:33:12.836] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher running: args [--gameDir, ., --launchTarget, fmlserver, --fml.forgeVersion, 26.0.63, --fml.mcpVersion, 20190603.175704, --fml.mcVersion, 1.14.2, --fml.forgeGroup, net.minecraftforge]
[22:33:12.839] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher starting: java version 1.8.0_212 by Oracle Corporation
[22:33:13.240] [main/INFO] [ne.mi.fm.lo.FixSSL/CORE]: Added Lets Encrypt root certificates as additional trust
[22:33:14.045] [main/INFO] [cp.mo.mo.LaunchServiceHandler/MODLAUNCHER]: Launching target 'fmlserver' with arguments [--gameDir, .]
[22:33:17.871] [main/INFO] [ne.mi.fm.ModLoader/CORE]: Loading Network data for FML net version: FML2
[22:33:28.974] [main/WARN] [minecraft/Commands]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[22:33:28.980] [main/WARN] [minecraft/Commands]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[22:33:28.985] [main/WARN] [minecraft/Commands]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[22:33:28.990] [main/WARN] [minecraft/Commands]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[22:33:28.995] [main/WARN] [minecraft/Commands]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[22:33:29.312] [Server thread/INFO] [minecraft/DedicatedServer]: Starting minecraft server version 1.14.2
[22:33:30.151] [modloading-worker-1/INFO] [ne.mi.co.ForgeMod/FORGEMOD]: Forge mod loading, version 26.0.63, for MC 1.14.2 with MCP 20190603.175704
[22:33:30.160] [modloading-worker-1/INFO] [ne.mi.co.MinecraftForge/FORGE]: MinecraftForge v26.0.63 Initialized
[22:33:30.418] [modloading-worker-2/FATAL] [ne.mi.fm.lo.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/gui/ScreenManager$IScreenFactory for invalid dist DEDICATED_SERVER
[22:33:30.427] [modloading-worker-2/ERROR] [ne.mi.fm.ja.FMLModContainer/LOADING]: Failed to create mod instance. ModID: essentials, class com.Da_Technomancer.essentials.Essentials
java.lang.RuntimeException: Attempted to load class net/minecraft/client/gui/ScreenManager$IScreenFactory for invalid dist DEDICATED_SERVER
	at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClass(RuntimeDistCleaner.java:71) ~[forge-1.14.2-26.0.63.jar:26.0] {}
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:59) ~[modlauncher-2.1.4.jar:?] {}
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:114) ~[modlauncher-2.1.4.jar:?] {}
	at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:236) ~[modlauncher-2.1.4.jar:?] {}
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:127) ~[modlauncher-2.1.4.jar:?] {}
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:97) ~[modlauncher-2.1.4.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_212] {}
	at java.lang.Class.getDeclaredMethods0(Native Method) ~[?:1.8.0_212] {}
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701) ~[?:1.8.0_212] {}
	at java.lang.Class.getDeclaredMethod(Class.java:2128) ~[?:1.8.0_212] {}
	at net.minecraftforge.eventbus.EventBus.getDeclMethod(EventBus.java:77) ~[eventbus-0.10.1-service.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.lambda$null$4(EventBus.java:89) ~[eventbus-0.10.1-service.jar:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_212] {}
	at java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1577) ~[?:1.8.0_212] {}
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_212] {}
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_212] {}
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:464) ~[?:1.8.0_212] {}
	at net.minecraftforge.eventbus.EventBus.lambda$registerObject$7(EventBus.java:91) ~[eventbus-0.10.1-service.jar:?] {}
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) ~[?:1.8.0_212] {}
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) ~[?:1.8.0_212] {}
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_212] {}
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:1.8.0_212] {}
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_212] {}
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418) ~[?:1.8.0_212] {}
	at net.minecraftforge.eventbus.EventBus.registerObject(EventBus.java:88) ~[eventbus-0.10.1-service.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.register(EventBus.java:114) ~[eventbus-0.10.1-service.jar:?] {}
	at com.Da_Technomancer.essentials.Essentials.<init>(Essentials.java:52) ~[?:1.14.2-2.1.1] {}
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_212] {}
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_212] {}
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_212] {}
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_212] {}
	at java.lang.Class.newInstance(Class.java:442) ~[?:1.8.0_212] {}
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:131) ~[?:26.0] {}
	at java.util.function.Consumer.lambda$andThen$0(Consumer.java:65) ~[?:1.8.0_212] {}
	at java.util.function.Consumer.lambda$andThen$0(Consumer.java:65) ~[?:1.8.0_212] {}
	at net.minecraftforge.fml.ModContainer.transitionState(ModContainer.java:112) ~[?:?] {}
	at net.minecraftforge.fml.ModList.lambda$null$9(ModList.java:122) ~[?:?] {}
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) [?:1.8.0_212] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) [?:1.8.0_212] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) [?:1.8.0_212] {}
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291) [?:1.8.0_212] {}
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731) [?:1.8.0_212] {}
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_212] {}
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_212] {}
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_212] {}
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0_212] {}
[22:33:30.526] [Server thread/FATAL] [ne.mi.fm.ModLoader/LOADING]: Failed to complete lifecycle event CONSTRUCT, 1 errors found
[22:33:30.527] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraftforge.fml.LoadingFailedException: null
	at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:177) ~[?:?] {}
	at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:159) ~[?:?] {}
	at net.minecraftforge.fml.server.ServerModLoader.begin(ServerModLoader.java:45) ~[?:?] {}
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:120) ~[?:?] {pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:586) [?:?] {pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212] {}
[22:33:30.559] [Server thread/ERROR] [minecraft/MinecraftServer]: This crash report has been saved to: /home/minecraft/./crash-reports/crash-2019-07-09_22.33.30-server.txt
[22:33:30.636] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server
[22:33:30.650] [Server thread/INFO] [minecraft/MinecraftServer]: Saving worlds
[22:33:30.654] [Server thread/ERROR] [minecraft/MinecraftServer]: Exception stopping the server
java.lang.IllegalArgumentException: Can not hotload overworld. This must be loaded at all times by main Server.
	at org.apache.commons.lang3.Validate.isTrue(Validate.java:158) ~[server-1.14.2-extra.jar:?] {}
	at net.minecraftforge.common.DimensionManager.initWorld(DimensionManager.java:201) ~[?:?] {}
	at net.minecraftforge.common.DimensionManager.getWorld(DimensionManager.java:170) ~[?:?] {}
	at net.minecraft.server.MinecraftServer.func_71218_a(MinecraftServer.java:962) ~[?:?] {pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_213211_a(MinecraftServer.java:509) ~[?:?] {pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_71260_j(MinecraftServer.java:541) ~[?:?] {pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.dedicated.DedicatedServer.func_71260_j(DedicatedServer.java:537) ~[?:?] {pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:646) [?:?] {pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212] {}
> root@tbot:/home/minecraft# 
```

**Comment by Da-Technomancer** [comment]
This is a known issue on servers, where dedicated servers with Essentials crash on startup but singleplayer and LAN worlds work fine. The versions of Essentials for 1.14.3+ have fixed this issue. 

Given that both vanilla Minecraft and Forge dropped support for MC1.14.2 the moment 1.14.3 was released, I don't provide support for 1.14.2 any more, and I'd personally suggest updating to MC1.14.3 asap, as bugfixes for vanilla, forge, and almost every mod on 1.14.x will only be on that version, and almost all 1.14.2 mods work cleanly on 1.14.3.

**Comment by Shanehar** [comment]
Turns out you are right. I was able to get it to run with the latest version. Then i learned this wasn't even what i was trying to install. Too many projects named Essentials. What i was trying to set up was Bukkit Essentials (Ended up going with Spigot EssentialsX) All the same, thank you for the fast reply. You were 100% correct and i appreciate your time.
